src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java

```diff
@@ -235,7 +245,9 @@ private AnomalyDetectorSettings() {}
     // TODO (kaituo): change to 4
     public static final int DEFAULT_MULTI_ENTITY_SHINGLE = 1;
 
-    public static final int MAX_SHINGLE_SIZE = 1024;
+    // max shingle size we have seen from external users
+    // the larger shingle size, the harder to fill in a complete shingle
+    public static final int MAX_SHINGLE_SIZE = 60;
```

Comment:
Makes sense. Does it make sense to bound the maximum dimensions directly to be 60x10 = 600? That is a bit less restrictive and even if people were using 100 features, they're ok as long as shingle size is not above 600/100.

Reply:
I understand we are more flexible if going this way.  But the change brings two complications:

1. dimension is a rcf concept.  For AD users, it is easier to have two concepts: max features and shingles.
2. the performance impact is unknown.  For example, previously, we query 10 features in one query; now we need to query 600 features in one query.  This needs to be evaluated.

Reply:
I think for the first the explanation is easy. I think some shrive like Kinesis sets the product to be 900 (or something similar). Basically, if we want large models we can be OOM -- that should be easy to explain.

For 2, If the user creates a 600 features in one query then that may be a bigger issue independent of AD :)

This seems like a two way door - if use cases come up, the change can be done easily.